var/spack/repos/builtin/packages/openfoam-com/package.py

```diff
@@ -659,6 +663,10 @@ def __init__(self, spec, **kwargs):
         # spec.architecture.target is like `uname -m`
         target   = spec.architecture.target
 
+        # Platform detects as 'cray' - treat like 'linux'
+        if platform == 'cray':
+            platform = 'linux'
```

Comment:
Sorry about the misunderstanding (don't mean to be obtuse). You are right, this is really ugly. If the cray information cannot be trusted, I guess we need either need to revert to the python platform solution that you had (although then I'd only like to use this for cray, since the others seem to ok as-is), and/or tag @mpbelhorn to see what his take on the situation is. He seems to be one for cray.